### DISCUSSION TITLE
How can I use specialized decoder options in Image.Identify

### DISCUSSION THREAD

**Discussion by andreas-eriksson** [SixLabors/ImageSharp#2655] [Q&A] [answered]
I want to ignore corrupt png files when running `Image.Identify`, but I cannot figure out how to do it.

The `Identify` method only accepts `DecoderOptions` and not `PngDecoderOptions`.

I wanted to do something like this but it does not compile.
`var imageInfo = Image.Identify(new PngDecoderOptions { PngCrcChunkHandling = PngCrcChunkHandling.IgnoreData }, stream);`

**Comment by JimBobSquarePants** [comment]
If you're just ignoring them, could you do a `try..catch`? We should make identify use the less strict parsing by default in future builds.

**Reply by andreas-eriksson** [reply]
Sorry, I meant that I want to ignore the CRC check so that I can retrieve some information about the image even if it is corrupt.

Is there any other way I can accomplish this? Can I change the default configuration to always ignore CRC errors for png?

**Reply by andreas-eriksson** [reply]
Any update on this?

**Reply by JimBobSquarePants** [reply, marked as answer]
Don't do that. This project source is open. What was stopping you having a look?

Currently `Identify` cannot be customized. The default options could perhaps be relaxed given that no data reading occurs but that will not happen before v4.

**Reply by andreas-eriksson** [reply]
I had a look but I couldn't figure out if it was indeed possible or not. Thanks for the info.

Would there be chance of making it possible before v4 if I contribute with a PR?
